analyticsdataserver/urls.py

```diff
@@ -24,5 +24,8 @@
     admin.autodiscover()
     urlpatterns.append(url(r'^site/admin/', include(admin.site.urls)))
 
+if  'enterprise_data' in settings.INSTALLED_APPS:
```

Comment:
But it will always be in `INSTALLED_APPS` according to your base settings file. If installing the `edx-enterprise-data` app needs to be optional, we should probably come up with a real way of doing that. Isn't this the problem that Nimisha solved for edx-platform?

Reply:
@nasthagiri ^^

Reply:
@douglashall Yes, the plugin architecture would allow this, but it is also currently coupled to edx-platform.

My thought here was that one could remove enterprise_data from the THIRD_PARTY_APPS and not have to update a bunch of other places such as this `urls.py` file

Reply:
But wouldn't you want them to remove this? Otherwise you end up with dead code.

Reply:
I see it as a configuration switch, just the configuration happens to be if the enterprise_data app is installed. If its not installed, this code will get skipped.

Reply:
FWIW, the newly created plugin architecture in edx-platform can be easily extracted into its own repo.  The fundamental design and implementation are generic so can apply to any other IDA.
There are a few Enums that specifically mention LMS and Studio - but those Enums can remain in edx-platform.  Let me know if you need further input if your team chooses to take on the extraction. Thanks.